app/scripts/lib/account-tracker.js

```diff
@@ -256,7 +256,8 @@ export default class AccountTracker {
     ethContract.balances(addresses, ethBalance, (error, result) => {
       if (error) {
         log.warn(`MetaMask - Account Tracker single call balance fetch failed`, error)
-        return Promise.all(addresses.map(this._updateAccount.bind(this)))
+        Promise.all(addresses.map(this._updateAccount.bind(this)))
```

Comment:
should there be an `await` here? the place where this function is called `await`s, but without return a promise, doesn't this function call become non-blocking?

or maybe I don't know something about async await...

Reply:
There should but that's a functional change for a different PR. This works no worse/better than before.

(Also this is a common problem in the codebase and we should audit all uses of promises `async`/`await` systematically.)

Reply:
but I mean, isn't `return Promise.all(addresses.map(this._updateAccount.bind(this)))` functionally different from
```
Promise.all(addresses.map(this._updateAccount.bind(this)))
return
```
?

Reply:
It's equivalent if the return value is unused. Since the function in question here is the callback being passed to `ethContract.balances`, the return value is not used. Nothing is `await`-ing this `Promise`.

Reply:
If the return value were used, yes, but it is not.

Reply:
But this is significantly easier change than ensuring the function continues to work when the timing semantics have changed. If we wait for the updates to happen that would be more different.

Reply:
" the function in question here is the callback being passed to"

doh, right, I should have realized that